src/hardware/setup.rs

```diff
@@ -133,6 +143,80 @@ pub struct PounderDevices {
     pub timestamper: pounder::timestamp::Timestamper,
 }
 
+impl PounderDevices {
+    pub fn update_dds(
+        &mut self,
+        settings: PounderConfig,
+        clocking: &mut ClockConfig,
+    ) {
+        if *clocking != settings.clock {
+            validate_clocking(
+                settings.clock.reference_clock,
+                settings.clock.multiplier,
+            )
+            .unwrap();
+
+            self.pounder
+                .set_ext_clk(settings.clock.external_clock)
+                .unwrap();
+
+            self.dds_output
+                .builder()
+                .set_system_clock(
+                    settings.clock.reference_clock,
+                    settings.clock.multiplier,
+                )
+                .unwrap()
```

Comment:
Won't this panic if the user provides invalid settings? We definitely don't want that. Same with above w.r.t. clock settings